src/rhsm/config.py

```diff
@@ -105,7 +105,7 @@ def in_container():
     """
     # For development in containers we must be able to turn container detection
     # off
-    if os.environ.get("SMDEV_CONTAINER_OFF", False):
+    if os.getenv("SMDEV_CONTAINER_OFF", "False") in ["True", "true"]:
```

Comment:
There is no need to use list. Tuple is enough. Original `strtobool` method contains more options in the list:

```python
('y', 'yes', 't', 'true', 'on', '1')
```

I can imagine that the tuple can be extended to something else like ("enabled", "yeah", ...).

It would be also good to convert result of `os.getenv("SMDEV_CONTAINER_OFF", "False")` to lower case using `lower()` method.

Reply:
Do you expect someone setting that env variable to t, 1  etc? Why make it complicated?

Reply:
If not set to True/true we're safe to ignore it IMO.

Reply:
I don't see any reason, why to restrict the list only to true/True.

Reply:
@rabi,

After some discussions with the Candelpin team, we may have a "better" and easier way by considering an "empty" value as "False". This would require 2 changes:
- in your patch, we can drop the _strtobool() and replace only one line:
```Diff
- if os.environ.get("SMDEV_CONTAINER_OFF", False):
+ if os.environ.get("SMDEV_CONTAINER_OFF", '') != '':
```
- in the ansible-role-redhat-subscription, the default value would then be an empty string instead of the current False.

Note that OSP isn't really affected by this issue, since we're running the role only on the hardware node, but it would indeed be cleaner to make things right with it.

What do you think? Would you agree with that?

Reply:
@cjeanner meh, I don't really understand the argument here... I still feel the env var is named as something which is understood as a boolean (though all env vars are strings). IMO, using empty string when unset is ambiguous. Having said that, there has been too much bike-shedding in this PR to my liking. As I  suggested earlier, someone can go ahead and update this PR (to whatever the repo owners like) or close it and create a new PR to fix the issue.